src/app/CommandHandler.h

```diff
@@ -71,6 +71,47 @@ class CommandHandler : public Command
         virtual bool CommandExists(const ConcreteCommandPath & aCommandPath) = 0;
     };
 
+    class Handle
+    {
+    public:
+        Handle() {}
+        Handle(const Handle & handle) = delete;
+        Handle(Handle && handle)
+        {
+            mpHandler        = handle.mpHandler;
+            mMagic           = handle.mMagic;
+            handle.mpHandler = nullptr;
+            handle.mMagic    = 0;
+        }
+        Handle(decltype(nullptr)) {}
+        Handle(CommandHandler * handle);
+        ~Handle() { Release(); }
+
+        Handle & operator=(Handle && handle)
+        {
+            Release();
+            mpHandler        = handle.mpHandler;
+            mMagic           = handle.mMagic;
+            handle.mpHandler = nullptr;
+            handle.mMagic    = 0;
+            return *this;
+        }
+
+        Handle & operator=(decltype(nullptr))
+        {
+            Release();
+            return *this;
+        }
+
+        CommandHandler * Get();
```

Comment:
Needs to document semantics, and that in particular it can start returning nullptr after returning a non-null value without any assignments or Release calls, just based on what is happening with the stack.